example-crates/wallet_esplora_async/src/main.rs

```diff
@@ -53,19 +54,22 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
             (k, k_spks)
         })
         .collect();
-    let update = client
-        .scan(
-            local_chain,
-            keychain_spks,
-            [],
-            [],
-            STOP_GAP,
-            PARALLEL_REQUESTS,
-        )
+    let (update_graph, last_active_indices) = client
+        .update_tx_graph(keychain_spks, None, None, STOP_GAP, PARALLEL_REQUESTS)
         .await?;
-    println!();
+    let get_heights = wallet.tx_graph().missing_blocks(wallet.local_chain());
+    let new_tip = client.update_local_chain(prev_tip, get_heights).await?;
```

Comment:
why can't this just return `local_chain::Update`?

Reply:
Good catch. Changed to return the beautiful `Update` structure as you have requested.